pyomo/core/plugins/transform/remove_zero_terms.py

```diff
@@ -0,0 +1,64 @@
+# -*- coding: UTF-8 -*-
+"""Transformation to remove zero terms from constraints."""
+from __future__ import division
+
+import textwrap
+
+from pyomo.core.base.constraint import Constraint
+from pyomo.core.plugins.transform.hierarchy import IsomorphicTransformation
+from pyomo.repn.canonical_repn import generate_canonical_repn
+from pyomo.util.plugin import alias
+
+__author__ = "Qi Chen <https://github.com/qtothec>"
+
+
+class RemoveZeroTerms(IsomorphicTransformation):
+    """Looks for 0 * var in a constraint and removes it.
```

Comment:
This transformation will need to be reworked to support Pyomo5 expressions.  Hence, I'm not going to provide a detailed review here.